### ISSUE TITLE
Improve Feldman Cousins tests

### ISSUE THREAD

**Issue by cdeil** [gammapy/gammapy#2176]
I noticed that `gammapy/stats/tests/test_feldman_cousins.py::test_numerical_confidence_interval_values` sometimes fails, because it draws random numbers without setting a seed, and the assert tolerance is too tight.

https://github.com/gammapy/gammapy/blob/8b74b772808ed8720f18896940325add0c911804/gammapy/stats/tests/test_feldman_cousins.py#L116-L148

To reproduce the fail:
```
pip install pytest-repeat
pytest gammapy/stats/tests/test_feldman_cousins.py::test_numerical_confidence_interval_values --count 100
```
Log: https://gist.github.com/cdeil/ccbc02b31e20d152e1ae3c85f4c1d83f

Generally we try to avoid random numbers in tests and just use a small fixed sample test case and run that. But in this case, I think maybe keeping the test as-is and just fixing the seed could be the way to go?

---

In `test_numerical_confidence_interval_pdfs` there are also random numbers. The test doesn't fail, but still I'd suggest to either get rid of the random number generation or to fix the seed.

I did have one case where I think the test was hanging and I aborted after ~ 1 minute, but now I can't reproduce that, runtime only varies a little bit. But especially for algorithms such as FC where runtime depends on input data, we really should avoid passing different test data on each test run. There can still be MC coverage studies, but those should be in scripts in the `dev` folder - not in automated tests that run all the time on developer and continuous integration systems, those should be simple and very fast (to test regressions in Numpy, Astropy, Gammapy, ...)

---

@dlennarz - Can you please comment what change to the test you prefer?
Do you have time to send a PR (in the coming weeks), or should I do it?

**Comment by dlennarz** [comment]
One of the main features of the numerical Feldman-Cousins solution is that you can use it without analytically knowing the underlying probability density function. That's why `test_feldman_cousins.py::test_numerical_confidence_interval_values` draws 5000 random numbers and tries to get to the correct upper limit.

It looks like 5000 random numbers is not enough to always reach the accuracy of 0.1 used by the test (in all your failed cases, the deviation was 0.11, so only barely outside). More random numbers would be needed to get a more accurate result, but that would increase the run time of the test.

The test should be fast and reproducible, so I think fixing the random seed it the cleanest solution. Increasing the tolerance could still mean there might be an occasional set of random numbers that doesn't reach the tolerance requested. Fixing the random seed also allows to reduce the amount of random numbers needed, thus speeding up the test.

There are no random numbers in `test_numerical_confidence_interval_pdfs`, it uses the probability mass function, which is fixed for each value of mu. There should be no variance in the runtime of this function.

Everything should be fixed with #2182.

**Comment by cdeil** [comment]
@dlennarz - Thanks!

I'll review and merge #2182 later today.
